ed25519/src/lib.rs

```diff
@@ -402,6 +408,62 @@ impl<'de> Deserialize<'de> for Signature {
     }
 }
 
+#[cfg(all(feature = "serde", feature = "serde_bytes"))]
+impl serde_bytes::Serialize for Signature {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_bytes(&self.0)
+    }
+}
+
+#[cfg(all(feature = "serde", feature = "serde_bytes"))]
+impl<'de> serde_bytes::Deserialize<'de> for Signature {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        struct ByteArrayVisitor;
+
+        impl<'de> de::Visitor<'de> for ByteArrayVisitor {
+            type Value = [u8; SIGNATURE_LENGTH];
+
+            fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+                formatter.write_str("bytestring of length 64")
+            }
+
+            fn visit_bytes<E>(self, bytes: &[u8]) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                use de::Error;
+
+                bytes
+                    .try_into()
+                    .map_err(|_| Error::invalid_length(bytes.len(), &self))
+            }
+
+            #[cfg(feature = "std")]
+            fn visit_byte_buf<E>(self, bytes: std::vec::Vec<u8>) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                use de::Error;
+
+                let bytes_len = bytes.len();
+                bytes
+                    .try_into()
+                    .map_err(|_| Error::invalid_length(bytes_len, &self))
```

Comment:
I think this might require bumping MSRV to 1.47, which is fine

Reply:
Ah yeah, looks like `TryFrom<Vec<T>>` was stabilised in 1.48: https://doc.rust-lang.org/stable/std/primitive.array.html#impl-TryFrom%3CVec%3CT%2C%20A%3E%3E. I can use a different approach, probably not worth bumping MSRV just for this.